site2/website/blog/2019-07-05-Apache-Pulsar-2-4-0.md

```diff
@@ -0,0 +1,118 @@
+---
+author: Penghui Li
+authorURL: https://twitter.com/lipenghui6
+title: Apache Pulsar 2.4.0
+---
+
+We are glad to present the new 2.4.0 release of Pulsar. This is the result of a huge 
+effort from the community, with over 460 commits and a long list of new features, 
+general improvements and bug fixes.
+
+Check out the <b>[release notes](/release-notes/#2.4.0)</b> for a detailed list of 
+the changes, with links to the relevant pull-requests, discussions and documentation.
+
+Regarding new features introduced, I just want to highlight here a tiny subset of them:
+
+<!--truncate-->
+
+### Delayed message delivery
+
+It's now possible to send delayed message by pulsar producer, and delayed message will 
+available after delay time.
+
+The Java code for a client using delayed message delivery will look like:
+
+```java
+producer.newMessage().value("delayed message").deliverAfter(10, TimeUnit.SECONDS).send()
+```
+
+> Note:
+>
+> 1. Messages are only delayed on shared subscriptions. Other subscriptions will deliver immediately.
+> 2. Can't work well with batching messages while messages has different delay time.
```

Comment:
Delayed message cannot work well with ...?

Reply:
> Can't work well with batching messages while messages has different delay time.

It does work. It will just send the messages individually.

Reply:
That's really great news.